scalableminds/webknossos
# Return HTTP error 401 for authorization errors

Until now, we redirected an unauthorised request to the dashboard / landing page which makes no sense in the context of an request API. Instead the server should return an HTTP 401 error and the client can handle that accordingly.

### Mailable description of changes:
- Prominently display an error notification when a user is logged out / not authorized to view a resource.

### Steps to test:
- Open wK in two tabs
- Log out in the first tab
- Try to navigate the some admin page in the seconds tab. While trying to download the required information for users / projects / etc an error notification should pop up. The view should not finish loading. (Permanent loading spinner)

### Questions:
1. Does this break anything? 
2. Is there a way in the client to handle 401 and act accordingly? The `Request` lib could handle all 401 errors, but that would mean a very tight integration with `redux` to logout the existing user and the react router for redirecting to the login view. Can we catch the error somewhere else instead? The router?

------
